crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -207,30 +182,23 @@ module sui::staking_pool {
             EWRONG_POOL
         );
 
-        assert!(delegation.principal_sui_amount == balance::value(&staked_sui.principal), EINSUFFICIENT_SUI_TOKEN_BALANCE);
```

Comment:
Pointing out for other reviews: the rest of the changes in the PR are very mechanical--I think almost everything tricky happens here